File: javascript/isPalindrome.js
/**
 * Given an integer x, return true if x is palindrome integer.
 * An integer is a palindrome when it reads the same backward as forward. 
 * For example, 121 is palindrome while 123 is not.
 * @param {number} x
 * @return {boolean}
 */
 var isPalindrome = function (x) {
    //  convert number into string
    const y = x.toString();
    // get length of string for readability
    const length = y.length;
    // funciton output
    let output = true;

    // loop through the string
    for (let index = 0; index < length; index++) {
        // get front and rear positions for read
        let front = index;
        let back = length - 1 - index;
        // if back and front are equal to each other (odd numbered string)
        // or back is great than front (even numbered string), break the loop and break
        if (back <= front) {
            break;
        } else {
            // if the front does not equal the backside, set to false and break
            if (y[front] != y[back]) {
                output = false;
                break;
            }
        }
    }
    return output;
};

console.log(isPalindrome(010101));

File: python/is_palindrome.py
'''
https://leetcode.com/problems/palindrome-number/

An integer is a palindrome when it reads the same backward as forward. 
For example, 121 is palindrome while 123 is not.
'''
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # convert the number into a string
        x_string = str(x)
        # grab length of string for readability
        length = len(x_string)
        # output of the funciton
        output = True

        # iterate through the string
        for x in range(0, length):
            # get the front and back index positions for readability
            front = x
            back = length - 1 - x
            # if back and front are equal to each other (odd numbered string)
            # or back is great than front (even numbered string), break the loop and break
            if back <= front:
                break
            else:
                # if the front does not equal the backside, set to false and break
                if x_string[front] != x_string[back]:
                    output = False
                    break
        return output

